File: C1.UWP.FlexChart/CS/FlexChartExplorer/Converters.cs
using System;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;
using C1.Chart;
using System.Collections.Generic;
using System.Linq;

namespace FlexChartExplorer
{
    public class BooleanConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return value.ToString();
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }

    public class EnumConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return value?.ToString();
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            string str = value.ToString();
            switch (parameter.ToString())
            {
                case "Palette":
                    return (Palette)Enum.Parse(typeof(Palette), str);
                case "Stacking":
                    return (Stacking)Enum.Parse(typeof(Stacking), str);
                case "PieLabelPosition":
                    return (PieLabelPosition)Enum.Parse(typeof(PieLabelPosition), str);
                case "LabelPosition":
                    return (LabelPosition)Enum.Parse(typeof(LabelPosition), str);
                case "ChartSelectionMode":
                    return (ChartSelectionMode)Enum.Parse(typeof(ChartSelectionMode), str);
                case "Position":
                    return (Position)Enum.Parse(typeof(Position), str);
                case "FitType":
                    return (FitType)Enum.Parse(typeof(FitType), str);
                case "QuartileCalculation":
                    return (QuartileCalculation)Enum.Parse(typeof(QuartileCalculation), str);
                case "ErrorAmount":
                    return (ErrorAmount)Enum.Parse(typeof(ErrorAmount), str);
                case "Direction":
                    return (ErrorBarDirection)Enum.Parse(typeof(ErrorBarDirection), str);
                case "EndStyle":
                    return (ErrorBarEndStyle)Enum.Parse(typeof(ErrorBarEndStyle), str);
                case "Orientation":
                    return (Orientation)Enum.Parse(typeof(Orientation), str);
                case "TextWrapping":
                    return (C1.Chart.TextWrapping)Enum.Parse(typeof(C1.Chart.TextWrapping), str);
                case "TreeMapType":
                    return (TreeMapType)Enum.Parse(typeof(TreeMapType), str);
                case "HistogramBinning":
                    return (HistogramBinning)Enum.Parse(typeof(HistogramBinning), str);
                case "LabelOverlapping":
                    return (LabelOverlapping)Enum.Parse(typeof(LabelOverlapping), str);
                case "DataShape":
                    return (DataShape)Enum.Parse(typeof(DataShape), str);
                case "OverlappingLabels":
                    return (OverlappingLabels)Enum.Parse(typeof(OverlappingLabels), str);
                case "GroupSeperator":
                    return (AxisGroupSeparator)Enum.Parse(typeof(AxisGroupSeparator), str);               
            }
            return null;
        }
    }

    public class ChartTypeConverter : DependencyObject, IValueConverter
    {
        public Dictionary<ChartType, string> ChartTypes
        {
            get { return (Dictionary<ChartType, string>)GetValue(ChartTypesProperty); }
            set { SetValue(ChartTypesProperty, value); }
        }

        public static readonly DependencyProperty ChartTypesProperty =
            DependencyProperty.Register("ChartTypes", typeof(Dictionary<ChartType, string>), typeof(ChartTypeConverter), 
            new PropertyMetadata(null));

        public object Convert(object value, Type targetType, object parameter, string language)
        {
            var key = (ChartType)value;
            int index = 0;
            foreach (var chartType in ChartTypes)
            {
                if (chartType.Key.Equals(key))
                    return index;
                index++;
            }

            return index;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            int index = (int)value;
            return ChartTypes.ElementAt(index).Key;
        }
    }

    public class GroupSeparatorConverter : DependencyObject, IValueConverter
    {
        public Dictionary<AxisGroupSeparator, string> GroupSeparator
        {
            get { return (Dictionary<AxisGroupSeparator, string>)GetValue(GroupSeparatorProperty); }
            set { SetValue(GroupSeparatorProperty, value); }
        }

        public static readonly DependencyProperty GroupSeparatorProperty =
            DependencyProperty.Register("GroupSeparator", typeof(Dictionary<AxisGroupSeparator, string>), typeof(GroupSeparatorConverter),
            new PropertyMetadata(null));

        public object Convert(object value, Type targetType, object parameter, string language)
        {
            var key = (AxisGroupSeparator)value;
            int index = 0;
            foreach (var type in GroupSeparator)
            {
                if (type.Key.Equals(key))
                    return index;
                index++;
            }

            return index;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            int index = (int)value;
            return GroupSeparator.ElementAt(index).Key;
        }
    }

    public class VisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            if (value == null)
                return Visibility.Collapsed;

            var position = (PieLabelPosition)Enum.Parse(typeof(PieLabelPosition), value.ToString());
            return position == PieLabelPosition.None ? Visibility.Collapsed : Visibility.Visible;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }

    public class OrderVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            var fitType = (FitType)Enum.Parse(typeof(FitType), value.ToString());
            return fitType == FitType.Fourier || fitType == FitType.Polynom ? Visibility.Visible : Visibility.Collapsed;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }

    public class BoolToVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            bool isChecked = (bool)value;
            return isChecked ? Visibility.Visible : Visibility.Collapsed;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }
}

File: C1.UWP.FlexChart/VB/FlexChartExplorer/Converters.vb
Imports System.Linq
Imports System.Collections.Generic
Imports C1.Chart
Imports Windows.UI.Xaml.Data
Imports Windows.UI.Xaml
Imports System

Public Class BooleanConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Return value.ToString()
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class

Public Class EnumConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Return value.ToString()
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Dim str As String = value.ToString()
        Select Case parameter.ToString()
            Case "Palette"
                Return CType([Enum].Parse(GetType(Palette), str), Palette)
            Case "Stacking"
                Return CType([Enum].Parse(GetType(Stacking), str), Stacking)
            Case "PieLabelPosition"
                Return CType([Enum].Parse(GetType(PieLabelPosition), str), PieLabelPosition)
            Case "LabelPosition"
                Return CType([Enum].Parse(GetType(LabelPosition), str), LabelPosition)
            Case "ChartSelectionMode"
                Return CType([Enum].Parse(GetType(ChartSelectionMode), str), ChartSelectionMode)
            Case "Position"
                Return CType([Enum].Parse(GetType(Position), str), Position)
            Case "FitType"
                Return CType([Enum].Parse(GetType(FitType), str), FitType)
            Case "ErrorAmount"
                Return CType([Enum].Parse(GetType(ErrorAmount), str), ErrorAmount)
            Case "Direction"
                Return CType([Enum].Parse(GetType(ErrorBarDirection), str), ErrorBarDirection)
            Case "EndStyle"
                Return CType([Enum].Parse(GetType(ErrorBarEndStyle), str), ErrorBarDirection)
            Case "Orientation"
                Return CType([Enum].Parse(GetType(Orientation), str), Orientation)
            Case "TextWrapping"
                Return CType([Enum].Parse(GetType(C1.Chart.TextWrapping), str), C1.Chart.TextWrapping)
            Case "HistogramBinning"
                Return CType([Enum].Parse(GetType(C1.Chart.HistogramBinning), str), C1.Chart.HistogramBinning)
        End Select
        Return Nothing
    End Function
End Class

Public Class ChartTypeConverter
    Inherits DependencyObject
    Implements IValueConverter
    Public Property ChartTypes() As Dictionary(Of ChartType, String)
        Get
            Return CType(GetValue(ChartTypesProperty), Dictionary(Of ChartType, String))
        End Get
        Set
            SetValue(ChartTypesProperty, Value)
        End Set
    End Property

    Public Shared ReadOnly ChartTypesProperty As DependencyProperty = DependencyProperty.Register("ChartTypes", GetType(Dictionary(Of ChartType, String)), GetType(ChartTypeConverter), New PropertyMetadata(Nothing))

    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim key As ChartType = CType(value, ChartType)
        Dim index As Integer = 0
        For Each chartType As KeyValuePair(Of ChartType, String) In ChartTypes
            If chartType.Key.Equals(key) Then
                Return index
            End If
            index += 1
        Next

        Return index
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Dim index As Integer = CType(value, Integer)
        Return ChartTypes.ElementAt(index).Key
    End Function
End Class

Public Class VisibilityConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        If value Is Nothing Then
            Return Visibility.Collapsed
        End If

        Dim position As PieLabelPosition = CType([Enum].Parse(GetType(PieLabelPosition), value.ToString()), PieLabelPosition)
        Return If(position = PieLabelPosition.None, Visibility.Collapsed, Visibility.Visible)
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class

Public Class OrderVisibilityConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim fitType As FitType = CType([Enum].Parse(GetType(FitType), value.ToString()), FitType)
        Return If(fitType = fitType.Fourier OrElse fitType = fitType.Polynom, Visibility.Visible, Visibility.Collapsed)
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class

Public Class BoolToVisibilityConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim isChecked As Boolean = CType(value, Boolean)
        Return If(isChecked, Visibility.Visible, Visibility.Collapsed)
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class